nan.h

```diff
@@ -1191,68 +1201,71 @@ NAN_INLINE _NanWeakCallbackInfo<T, P>* NanMakeWeakPersistent(
+    NAN_INLINE                                                                 \
+    v8::Local<v8::Value> Nan ## NAME(v8::Handle<v8::String> errmsg) {          \
+      return v8::Exception::NAME(errmsg);                                      \
+    }                                                                          \
+                                                                               \
+    NAN_INLINE v8::Local<v8::Value> NanThrow ## NAME(const char *errmsg) {     \
+      NanEscapableScope();                                                     \
+      return NanEscapeScope(NanNew(v8::ThrowException(                         \
+          v8::Exception::NAME(NanNew(errmsg)))));                              \
+    }                                                                          \
+                                                                               \
+    NAN_INLINE                                                                 \
+    v8::Local<v8::Value> NanThrow ## NAME(v8::Handle<v8::String> errmsg) {     \
+      NanEscapableScope();                                                     \
+      return NanEscapeScope(NanNew(v8::ThrowException(                         \
+          v8::Exception::NAME(errmsg))));                                      \
+    }
 
-# define _NAN_THROW_ERROR(fun, errmsg)                                         \
-    do {                                                                       \
-      NanScope();                                                              \
-      return v8::Local<v8::Value>::New(                                        \
-        v8::ThrowException(_NAN_ERROR(fun, errmsg)));                          \
-    } while (0);
+  ERROR_TYPES
 
-  NAN_INLINE v8::Local<v8::Value> NanError(const char* errmsg) {
-    return _NAN_ERROR(v8::Exception::Error, errmsg);
-  }
+# undef X
+# undef ERROR_TYPES
 
-  NAN_INLINE v8::Local<v8::Value> NanThrowError(const char* errmsg) {
-    _NAN_THROW_ERROR(v8::Exception::Error, errmsg);
-  }
 
-  NAN_INLINE v8::Local<v8::Value> NanThrowError(
-      v8::Handle<v8::Value> error
-  ) {
-    NanScope();
-    return v8::Local<v8::Value>::New(v8::ThrowException(error));
+  NAN_INLINE v8::Local<v8::Value> NanThrowError(v8::Handle<v8::Value> error) {
+    NanEscapableScope();
+    return NanEscapeScope(v8::Local<v8::Value>::New(v8::ThrowException(error)));
   }
 
-  NAN_INLINE v8::Local<v8::Value> NanError(
+  namespace Nan { namespace imp {
+    NAN_INLINE v8::Local<v8::Value> E(const char *msg, const int errorNumber) {
+      NanEscapableScope();
+      v8::Local<v8::Value> err = v8::Exception::Error(NanNew<v8::String>(msg));
+      v8::Local<v8::Object> obj = err.As<v8::Object>();
+      obj->Set(NanNew<v8::String>("code"), NanNew<v8::Int32>(errorNumber));
+      return NanEscapeScope(err);
+    }
+  }  // end of namespace imp
+  }  // end of namespace Nan
```

Comment:
They ought to be in different ifdefs, one for 0.10 and one for 0.12.

On May 8, 2015 10:27:08 PM EEST, Ben Noordhuis <notifications@github.com> wrote:
>>    }
>>  
>> -  NAN_INLINE v8::Local<v8::Value> NanError(
>> +  namespace Nan { namespace imp {
>> +    NAN_INLINE v8::Local<v8::Value> E(const char *msg, const int
>errorNumber) {
>> +      NanEscapableScope();
>> +      v8::Local<v8::Value> err =
>v8::Exception::Error(NanNew<v8::String>(msg));
>> +      v8::Local<v8::Object> obj = err.As<v8::Object>();
>> +      obj->Set(NanNew<v8::String>("code"),
>NanNew<v8::Int32>(errorNumber));
>> +      return NanEscapeScope(err);
>> +    }
>> +  }  // end of namespace imp
>> +  }  // end of namespace Nan
>
>This duplicates the E function above, doesn't it?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/iojs/nan/pull/323/files#r29969280